### DISCUSSION TITLE
Running uvicorn through terminal gives keepalive ping timeout error

### DISCUSSION THREAD

**Discussion by ChaitanyaYeole02** [Kludex/uvicorn#1660] [Potential Issue]
I am trying to run my program through this command. I have tried using 0.0 and also 0 in my ping interval and timeout. Still, I get this error.
Command Used
`uvicorn app:central_system --host "0.0.0.0" --port 9000 --header "Sec-WebSocket-Protocol":"ocpp1.6" --workers 1 --timeout-keep-alive 30 --log-level "debug" --ws-ping-interval 0.0 --ws-ping-timeout 0.0`

Error Received
```
←[36mDEBUG←[0m:    > CLOSE 1011 (unexpected error) keepalive ping timeout [24 bytes]
[write_frame_sync - 1158] > CLOSE 1011 (unexpected error) keepalive ping timeout [24 bytes]
```

Also, when I do it through my Python IDE (Python3.10). It works properly
```
if __name__ == '__main__':
    headers = [("Sec-WebSocket-Protocol", "ocpp1.6")]
    uvicorn.run("app:central_system", host="0.0.0.0", port=9000, headers=headers, workers=1, ws_ping_interval=None,
                ws_ping_timeout=None, use_colors=True, log_level='debug', timeout_keep_alive=30)
```

**Comment by euri10** [comment]
long time I havn't touched websockets and not sure what your clients expects, but if you disable ping pong on the server I guess that's expected to receive those logs, the 2nd programmatic use where you receive nothing worries me more, not sure passing None is valid, would have to check

**Reply by ChaitanyaYeole02** [reply]
There are many clients, which do not expect PING PONG. Also, WebSockets allows you to disable ping pong by passing None. The 2nd one is valid as it is a feature in WebSockets.

**Reply by euri10** [reply]
ok I see you pass 0.0, and that's afaik a click limitation where you can't pass None, 0 means you're basically asking for an instant pong, would a negative value circumvent that, if not that would be a limitation I suppose yes

**Reply by ChaitanyaYeole02** [reply]
I don't think it is asking for instant pong. We can disable the ping and pong behaviour. This is what I know so far. You can just read the start of the documentation once.
https://websockets.readthedocs.io/en/stable/reference/common.html#:~:text=Set%20ping_interval%20to%20None%20to,None%20to%20disable%20this%20behavior.

**Reply by euri10** [reply]
`None != 0.0`

**Reply by ChaitanyaYeole02** [reply]
Ok, so what can I enter into the command which equals to None?

**Reply by euri10** [reply]
I anwsered above, thanks
